manifests/server/config.pp

```diff
@@ -102,6 +102,10 @@
     Exec['puppet_server_config-generate_ca_cert'] ~> Service[$puppet::server_httpd_service]
   }
 
+  if $::puppet::server_implementation == 'puppetserver' and $::puppet::server_ca {
+    Exec['puppet_server_config-generate_ca_cert'] ~> Service['puppetserver']
```

Comment:
I think this may be superfluous, since puppet::server already notifies from puppet::server::config to puppet::server::service.  The httpd one is probably required because puppet::service::service contains only the puppetmaster/puppetserver service and not Apache.

It might be that the tests don't pick up the notify, but it should work as-is.